### ISSUE TITLE
Stopped working?

### ISSUE THREAD

**Issue by custa1200** [shenxn/ha-dyson#180]
My integrations seems to have stopped talking to my Dysons after updating to the latest HACS

**Comment by custa1200** [comment]
<img width="611" alt="CleanShot 2023-01-25 at 16 36 21@2x" src="https://user-images.githubusercontent.com/161224/214488488-31696be6-84c3-4016-a343-da8a2bbfb97e.png">
<img width="1478" alt="CleanShot 2023-01-25 at 16 36 33@2x" src="https://user-images.githubusercontent.com/161224/214488494-968ef6bc-9592-422d-a29a-986ed0845a3e.png">

**Comment by zzznz27** [comment]
Just started getting the same issue today too

**Comment by GitPetri** [comment]
Uh, good to know. Which version of HACS broke the integration? Running 1.28.0 at the moment and HA shows 1.30.0 as available one

**Comment by zzznz27** [comment]
I'm on 1.29. 1


-------- Original message --------
From: GitPetri ***@***.***>
Date: 25/1/23 7:45 pm (GMT+10:00)
To: shenxn/ha-dyson ***@***.***>
Cc: Zach Herb ***@***.***>, Comment ***@***.***>
Subject: Re: [shenxn/ha-dyson] Stopped working? (Issue #180)


Uh, good to know. Which version of HACS broke the integration? Running 1.28.0 at the moment and HA shows 1.30.0 as available one

**Comment by GitPetri** [comment]
> I'm on 1.29. 1

Thanks, staying in 1.28 for now then.

**Comment by custa1200** [comment]
I updated to 1.30 from 1.28

**Comment by zzznz27** [comment]
Went back to 1.28.4 and it's still broken

**Comment by zzznz27** [comment]
Google integration is busted too...
Might be a change in Dyson's end?

**Comment by GitPetri** [comment]
> Google integration is busted too... Might be a change in Dyson's end?

I am running 1.28.0 and my integration seems to be working at the moment

**Comment by Trabbi1999** [comment]
I'm trying to integrate my Humidify and Cool Formaldehyde but it says "Failed to connect" when I try to set up manually. Homebridge can connect to my Dyson, maybe this is also related to the issue.

**Comment by arc72** [comment]
Broken for me too since today....

2023-01-26 12:01:11.454 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration dyson_cloud which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-01-26 12:02:16.119 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration dyson_cloud: Unable to import component: cannot import name 'DysonPurifierHumidifyCoolFormaldehyde' from 'libdyson' (/usr/local/lib/python3.10/site-packages/libdyson/__init__.py)

**Comment by BePalm2022** [comment]
I just try to install the integration...but is is not found and not showing up

There is not even Dyson integration
![hasst](https://user-images.githubusercontent.com/114201788/215252350-aa5514e7-fb6b-4776-8ba6-f768fececd97.JPG)




nor there is a HACS ...
![hacst](https://user-images.githubusercontent.com/114201788/215252343-9771a94a-cc7d-47c8-aef8-b3c53fc56cc3.JPG)

anything I have to set to see it ...
located in Malaysia... should not play a role to see all integrations..

**Comment by idfxken** [comment]
I can´t connect with my serial number and credentials either, possibly dyson sent out  a firmware update?

Edit: Dyson Pure+Cool humidifier here

**Comment by nerajchand** [comment]
I've been having a similar issue since the [Home Assistant OS 9.5 update](https://github.com/home-assistant/operating-system/releases/tag/9.5) .

What i had to do to get it working again was update the "requirements" in the manifest.json to 
`"requirements": ["git+https://github.com/jezzab/libdyson.git@main#libdyson==0.9.0"],`

So the full manifest.json looks like:

```
{
    "domain": "dyson_local",
    "name": "Dyson Local",
    "config_flow": true,
    "documentation": "https://github.com/shenxn/ha-dyson",
    "issue_tracker": "https://github.com/shenxn/ha-dyson/issues",
    "dependencies": ["zeroconf"],
    "codeowners": ["@shenxn"],
    "requirements": ["git+https://github.com/jezzab/libdyson.git@main#libdyson==0.9.0"],
    "version": "0.16.4-4",
    "iot_class": "local_polling"
}
```

This is the same for dyson_local & dyson_cloud.
After a reboot, both of my Dyson Purifiers came back. 

Hope that helps :)

**Comment by idfxken** [comment]
That only fixes the invalid handler bug on install/updates. 
Not this, integration was working, but something else broke.

**Comment by idfxken** [comment]
Edit : Apologies, it seems that I messed up with the dyson password. Just checked with resniffing the mqtt connection from dyson.

**Comment by mb0686** [comment]
The same issue occured for me in the past few days. Neither this integration nor the Dyson app were working. Eventually I just unplugged the Dyson and now it's back up and running. I didn't change anything in Home Assistant.

**Comment by aLTeReGo-SWI** [comment]
@nerajchand thank you for the solution. That fixed the problem perfectly! 

> I've been having a similar issue since the [Home Assistant OS 9.5 update](https://github.com/home-assistant/operating-system/releases/tag/9.5) .
> 
> What i had to do to get it working again was update the "requirements" in the manifest.json to `"requirements": ["git+https://github.com/jezzab/libdyson.git@main#libdyson==0.9.0"],`
> 
> So the full manifest.json looks like:
> 
> ```
> {
>     "domain": "dyson_local",
>     "name": "Dyson Local",
>     "config_flow": true,
>     "documentation": "https://github.com/shenxn/ha-dyson",
>     "issue_tracker": "https://github.com/shenxn/ha-dyson/issues",
>     "dependencies": ["zeroconf"],
>     "codeowners": ["@shenxn"],
>     "requirements": ["git+https://github.com/jezzab/libdyson.git@main#libdyson==0.9.0"],
>     "version": "0.16.4-4",
>     "iot_class": "local_polling"
> }
> ```
> 
> This is the same for dyson_local & dyson_cloud. After a reboot, both of my Dyson Purifiers came back.
> 
> Hope that helps :)

**Comment by custa1200** [comment]
> The same issue occured for me in the past few days. Neither this integration nor the Dyson app were working. Eventually I just unplugged the Dyson and now it's back up and running. I didn't change anything in Home Assistant.

yeah I unlogged and plugged mine back in and it became available again .

**Comment by custa1200** [comment]
The issue seems to pop up intermittently still.

**Comment by dotvezz** [comment]
Unfortunately this integration is no longer maintained, but I've created a new fork and am actively working on the new Humidifier and Formaldehyde models. Have a look at #185 for more information, and let me know if you have any questions.

The forks do incorporate the latest changes to libdyson and a few other improvements, maybe it will help people who are dealing with this issue!
